.github/workflows/noir-js.yml

```
@@ -0,0 +1,53 @@
+name: Release and Publish Noir Js
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version number"
+        required: false
+
+jobs:
+  release-noir-js:
+    name: Release and Publish Noir Js
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Bump version
+        working-directory: ./tooling/noir_js
+        id: bump_version
+        run: |
+          if [ -z "${{ github.event.inputs.version }}" ]; then
+            NEW_VERSION=$(npm version patch --no-git-tag-version)
+          else
+            NEW_VERSION=$(npm version ${{ github.event.inputs.version }} --no-git-tag-version)
+          fi
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+
+      - name: Install dependencies
+        working-directory: ./tooling/noir_js
+        run: npm install
+
+      - name: Build noir-source-resolver
+        working-directory: ./tooling/noir_js
+        run: npm run build
+
+      - name: Publish to NPM
+        working-directory: ./tooling/noir_js
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+
+      - name: Configure git
+        run: |
+          git config user.name kevaundray
+          git config user.email kevtheappdev@gmail.com
+
+      - name: Commit updates
+        run: |
+          git add tooling/noir_js/package-lock.json
```

```suggestion
          git add yarn.lock
```

We shouln't have `package-lock.json` ever.